Purchases/Public/Purchases.swift

```diff
@@ -943,14 +943,16 @@ public extension Purchases {
      */
     @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
     @objc(purchaseProduct:withDiscount:completion:)
-    func purchase(product: SKProduct, discount: SKPaymentDiscount, completion: @escaping PurchaseCompletedBlock) {
-        let payment = storeKitWrapper.payment(withProduct: product, discount: discount)
-        purchase(product: product, payment: payment, presentedOfferingIdentifier: nil, completion: completion)
+    func purchase(product: SKProduct, discount: StoreProductDiscount, completion: @escaping PurchaseCompletedBlock) {
```

Comment:
How do we feel about changing the public API without a deprecation? I handled that in #1160. Maybe with that one we can include both type changes in the same rename so it should be good.

Reply:
Ahhh! Like... maybe this should be previous method should be obsoleted instead of deprecated? It's a different method with a similar signature so can we obsolete it? 🤔

Reply:
Ohhhhh, the types aren't taken into account in the signature? 😬 It's a major version bump so I _think_ an obsoletion is ideal here because we don't want the user to deal with `SKPaymentDiscount`s anymore. And I don't think they can because we remove that part of the API as well.

Reply:
Thanks, thanks helps. Otherwise we wouldn't be able to keep this implementation 👍